File: compiler-design/module_2/lab_2_4_karanik/src/Token.java
public class Token {

    private String value;
    private DomainTag tag;
    private FragmentPosition fragmentPosition;

    public Token(String value, DomainTag tag, FragmentPosition fragmentPosition) {
        this.value = value;
        this.tag = tag;
        this.fragmentPosition = fragmentPosition;
    }

    public String getValue() {
        return value;
    }

    public DomainTag getTag() {
        return tag;
    }


    public FragmentPosition getFragmentPosition() {
        return fragmentPosition;
    }
}

File: compiler-design/module_1/lab_1_4_karanik/src/main/kotlin/Token.kt
sealed class Token(val domainTag: DomainTag, val fragmentPosition: FragmentPosition) {
    class Identifier(fragmentPosition: FragmentPosition, val name: String, val code: Int)
        : Token(DomainTag.IDENTIFIER, fragmentPosition)
    class NumericLiteral(fragmentPosition: FragmentPosition, val value: Long)
        : Token(DomainTag.NUMERIC_LITERAL, fragmentPosition)
    class Keyword(fragmentPosition: FragmentPosition, val name: String)
        : Token(DomainTag.KEYWORD, fragmentPosition)
    class OperationSign(fragmentPosition: FragmentPosition, val name: String)
        : Token(DomainTag.OPERATION_SIGN, fragmentPosition)
    class StringLiteral(fragmentPosition: FragmentPosition, val name: String)
        : Token(DomainTag.STRING_LITERAL, fragmentPosition)
}